 03-el.c-MAXPATHLEN.diff by Gergely Nagy <algernon@debian.org>

 on systems without MAXPATHLEN, allocate memory dynamically

--- a/libedit/el.c	2009-06-23 08:57:04.000000000 +1000
+++ b/libedit/el.c	2009-06-23 08:57:30.000000000 +1000
@@ -481,7 +481,11 @@ el_source(EditLine *el, const char *fnam
 	if (fname == NULL) {
 #if 1
 		static const char elpath[] = "/.editrc";
+#ifdef MAXPATHLEN
 		char path[MAXPATHLEN];
+#else
+		char *path;
+#endif
 
 #ifndef __GLIBC__
 		if (issetugid())
@@ -489,10 +493,19 @@ el_source(EditLine *el, const char *fnam
 #endif
 		if ((ptr = getenv("HOME")) == NULL)
 			return (-1);
+#ifndef MAXPATHLEN
+                path = (char *)malloc(strlen(ptr) + strlen(elpath) + 10);
+		if (path == NULL)
+			return (-1);
+		/* Always succeeds, since we have enough memory */
+		strcpy(path, ptr);
+		strcat(path, elpath);
+#else
 		if (strlcpy(path, ptr, sizeof(path)) >= sizeof(path))
 			return (-1);
 		if (strlcat(path, elpath, sizeof(path)) >= sizeof(path))
 			return (-1);
+#endif
 		fname = path;
 #else
 		/*
